### PR TITLE
Add Designate contract

### PR DESCRIPTION
**Pull request by KickSeason**
Add designate contract so that we can add more roles in the future.

 * Add `DesignateContract`
 * Move oracle node list from `OracleContract` to `DesignateContract`
 * Add `StateValidator` node list

@erikzhang @shargon @Tommo-L

### CODE REVIEW

**Review by vncoelho — COMMENTED** [review]
Looks a good design for designating roles on the blockchain, such as the idea of running stateroots on other nodes.
Perhaps this is a good generic change even if we keep stateroort on CN.

File: src/neo/SmartContract/Native/DesignateContract.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+namespace Neo.SmartContract.Native
+{
+    [Flags]
+    public enum Role : byte
+    {
+        StateValidator = 0b00000001,
+        Oracle = 0b00000010
+    }
+
+    public sealed class DesignateContract : NativeContract
```
Comment (shargon) [inline src/neo/SmartContract/Native/DesignateContract.cs:22]: `AccessControlContract`?

**Comment by shargon** [comment]
I like the idea, it's more extensible.

File: src/neo/SmartContract/Native/DesignateContract.cs (hunk 1 above)
Comment (KickSeason) [reply src/neo/SmartContract/Native/DesignateContract.cs:22]: Why call `AccessControl`?
This is a contract that committee designate some nodes as some role.

File: src/neo/SmartContract/Native/DesignateContract.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+using System.Linq;
+
+namespace Neo.SmartContract.Native
+{
+    [Flags]
+    public enum Role : byte
+    {
+        StateValidator = 0b00000001,
+        Oracle = 0b00000010
+    }
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/DesignateContract.cs:20]: One file one class.

Comment (KickSeason) [reply src/neo/SmartContract/Native/DesignateContract.cs]: done

**Review by erikzhang — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**